app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt

```diff
@@ -66,37 +83,69 @@ class TransactionListViewModel
         }
     }
 
-    private fun getTransactions(
+    private suspend fun getTransactions(
         safe: Safe,
         safes: List<Safe>,
         owners: List<Owner>,
         type: TransactionPagingSource.Type
     ): Flow<PagingData<TransactionView>> {
 
+        var txLocal: TransactionLocal? = null
+        if (type == TransactionPagingSource.Type.QUEUE ) {
+            txLocal = transactionLocalRepository.updateLocalTxLatest(safe)
+        }
+
         val safeTxItems: Flow<PagingData<TransactionView>> = transactionsPager.getTransactionsStream(safe, type)
             .map { pagingData ->
                 pagingData
                     .map { txListEntry ->
                         when (txListEntry) {
                             is TxListEntry.Transaction -> {
-                                val isConflict = txListEntry.conflictType != ConflictType.None
-                                val txView =
-                                    getTransactionView(
-                                        chain = safe.chain,
-                                        transaction = txListEntry.transaction,
-                                        safes = safes,
-                                        needsYourConfirmation = txListEntry.transaction.canBeSignedByAnyOwner(owners),
-                                        isConflict = isConflict,
-                                        localOwners = owners
-                                    )
-                                if (isConflict) {
-                                    TransactionView.Conflict(txView, txListEntry.conflictType)
-                                } else txView
+                                // conflict is resolved if there is local tx with same nonce
+                                // that was submitted for execution
+                                if (txLocal?.safeTxNonce == txListEntry.transaction.executionInfo?.nonce) {
+                                    if (txLocal?.safeTxHash == txListEntry.transaction.id.split("_").last() && txListEntry.transaction.txStatus == TransactionStatus.AWAITING_EXECUTION) {
```

Comment:
I noticed that the code here depends on the internal structure of the transaction id. This should not be the case, since the id might change in the future, thus breaking this logic.

In addition, this was one of the API agreements.

Reply:
> there is no way to get safeTxHash from the transaction list response
maybe locally stored data could include the transaction's safeTxHash, backend transaction id, and ethereum tx hash?

this way, you store the data at the point of transaction submission, and then you can lookup the safeTxHash by the backend transaction id

Reply:
> api agreements

this was not documented but implied; besides, it's a good design practice

Reply:
> this way, you store the data at the point of transaction submission, and then you can lookup the safeTxHash by the backend transaction id 

having to request tx details just to get safeTxHash would be inefficient. Better to stick with timestamps for now.

Reply:
oh no, what I mean is you store the safeTxHash from the transaction details when you execute/submit the transaction - you have that data and the transaction id beforehand. So, if you store it in advance, then you can look it up later when searching by the transaction id. No need for additional network requests